src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java

```diff
@@ -528,6 +531,8 @@ private void respondWith(HttpObject httpObject) {
     private void connectAndWrite(HttpRequest initialRequest) {
         LOG.debug("Starting new connection to: {}", remoteAddress);
 
+        this.clientConnection.channel.attr(remoteAddressAttrKey).set(remoteAddress);
+
```

Comment:
@mjallday each time a connection is established a new `ClientToProxyConnection` class is created with its corresponding new channel here https://github.com/adamfisk/LittleProxy/blob/master/src/main/java/org/littleshoot/proxy/impl/DefaultHttpProxyServer.java#L514
https://github.com/adamfisk/LittleProxy/blob/master/src/main/java/org/littleshoot/proxy/impl/ProxyConnection.java#L591
Also I read a book about netty and there was nothing about cleaning up attributes in the channel, it was said that the attributes are a good way of sharing state between components that may use the same channel (as it is thread safe, it can determine if the channel was called from particular thread and if not put it in queue)

But let me clarify it further so I am 100 percent sure